src/core/Provider.ts

```diff
@@ -132,7 +132,7 @@ const replaceDependencies = (
 ) => {
   const dependencies = new Set(dependenciesToReplace)
   let nextState = prevState
-  mKeys(nextState).forEach((a) => {
+  mForEach(nextState, (a) => {
     const aState = mGet(nextState, a) as AtomState<unknown>
```

Comment:
Oh, I missed this. I wanted to avoid `mGet` too.

```suggestion
  mForEach(nextState, (aState, a) => {
```

We would like to this. callback is `(value, key) => ...`. cf. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach